### DISCUSSION TITLE
Environments for private repositories

### DISCUSSION THREAD

**Discussion by benoitchantre** [community/community#25221] [Actions]
<aside class="onebox allowlistedgeneric" data-onebox-src="https://github.blog/changelog/2021-06-22-github-actions-environments-environment-protection-rules-and-environment-secrets-are-generally-available/">
  <header class="source">
      <img src="https://github.blog/wp-content/uploads/2019/01/cropped-github-favicon-512.png?fit=32%2C32" class="site-icon" width="32" height="32">

      <a href="https://github.blog/changelog/2021-06-22-github-actions-environments-environment-protection-rules-and-environment-secrets-are-generally-available/" target="_blank" rel="noopener nofollow ugc">The GitHub Blog</a>
  </header>

  <article class="onebox-body">
    <div class="aspect-image" style="--aspect-ratio:690/362;"><img src="https://github.blog/wp-content/uploads/2019/03/blog-card.png?fit=5001%2C2626" class="thumbnail" width="690" height="362"></div>

<h3><a href="https://github.blog/changelog/2021-06-22-github-actions-environments-environment-protection-rules-and-environment-secrets-are-generally-available/" target="_blank" rel="noopener nofollow ugc">GitHub Actions: Environments, environment protection rules, and environment...</a></h3>

  <p>GitHub Actions: Environments, environment protection rules, and environment secrets are generally available</p>

  </article>

  <div class="onebox-metadata">
    
    
  </div>

  <div style="clear: both"></div>
</aside>

<p>Does that mean environments are available to private repositories for free accounts?</p>
<p>If yes, the documentation should be updated.<br>
If no, what do you mean by “generally”?</p>
<p>Thanks for the clarification.</p>

**Comment by korshavn** [comment]
<p>Same question for private repositories in a Pro plan. Have a lot of them and I see no Environments available still…<br>
I have been waiting for the environments feature, and surely expected that “generally available” would mean available for also private repositories.</p>

**Comment by crash7** [comment]
<p>Same boat here, been waiting for a long time already and they seem to forget “private repositories”…</p>

**Comment by Zerotask** [comment, marked as answer]
<p>generally means that it’s out of the development phase (a beta for example).</p>
<p>If you don’t see it with a free plan, this means it’s not available for free tier.<br>
I have a pro subscription and therefore I can’t test it. I do have the option to set env variables in private repos.</p>

**Comment by harpstertech** [comment]
<p>It appears to be available only for enterprise and for public repos. I seriously hope they reconsider and make this available for private repos on Teams and other plans. It is pretty core, bare bones CICD necessities.</p>

**Comment by rodt7** [comment]
<p>Same here as a paid pro user. Really hope we could use this environment feature ASAP. It’s already 2022.</p>

**Comment by chrisfgonzales** [comment]
<p>This is really frustrating. I’m having the same issue. I don’t need all of the GitHub Enterprise features for the current project. I just need to be able to define a development and a production environment.</p>
<p>I don’t know why this was closed as resolved.</p>

**Comment by ryanhelms** [comment]
<p>It’s incredible to me that 2 years later this is still a thing. Get with it GitHub, release environments for all users who pay for your service. Of all providers, I would expect that you understand the necessity of environments from a fundamental level during software development. Sounds like it is time to find another provider who’s more in tune with the development cycle.</p>

**Comment by hholst80** [comment]
<p>I am looking into <a href="http://GitLab.com" rel="noopener nofollow ugc">GitLab.com</a> just because of this thing.</p>

**Comment by mihma** [comment]
I could not believe my eyes when I learned that it's GH Enterprise feature only. Seriously it should be enabled on GH Teams plan as it's a core CI/CD feature.

**Comment by Olsenius** [comment]
Im not able to find the https://docs.github.com/en/actions/deployment/targeting-different-environments/using-environments-for-deployment#environment-protection-rules on environments created on private repositories on our paid Teams plan. Is this a feature that require the enterprise plan?

**Reply by crash7** [reply]
Yes, sadly 😔 


(the comment just above yours actually says that).

**Comment by benoitchantre** [comment]
Finally 😀

> We are excited to announce that private repositories in Team and Pro plans now have access to GitHub Actions environments, deployment branches, and environment secrets everywhere—in the past, only teams on the Enterprise plan had access to all these features with private repositories.

Source : https://github.blog/2022-08-10-github-pages-now-uses-actions-by-default/

**Reply by tobi-usln** [reply]
@benoitchantre But this means that only limiting deployment protection as well as environment secrets are available, right? Required reviewers and Wait Timer is still not available, correct?

**Reply by benoitchantre** [reply]
@tobi-usln Now, with a Team account, you can a manage environments from the UI including their secrets, but required reviewers and wait timer need an Enterprise account.
